Mathlib/Init/Algebra/Order.lean

```diff
@@ -232,6 +238,10 @@ class LinearOrder (α : Type u) extends PartialOrder α, Min α, Max α :=
   min_def : ∀ a b, min a b = if a ≤ b then a else b := by intros; rfl
   /-- The minimum function is equivalent to the one you get from `maxOfLe`. -/
   max_def : ∀ a b, max a b = if a ≤ b then b else a := by intros; rfl
+  compare a b := compareOfLessAndEq a b
+  /-- Comparison via `compare` is equal to the canonical comparison given decidable `<` and `=`. -/
+  compare_eq_compareOfLessAndEq : ∀ a b, compare a b = compareOfLessAndEq a b := by
```

Comment:
Do we have any lemmas that say `compare a b = lt \iff a < b` etc?

Reply:
Ooh, right, we should have those. Assuming I didn't miss them (quite possible), I'd like to add them somewhere. If so I'll do so tomorrow.

Reply:
Okay, I've added `compare_lt_iff_lt`, `compare_gt_iff_gt`, `compare_eq_iff_eq`, and `compare_trichotomy`. Are there any others you can think of that we should have?